library.go

```diff
@@ -1 +1,31 @@
 package mc_router
+
+import (
+	"strings"
+)
+
+// SplitExternalHosts splits a string containing external hostnames by comma and/or newline delimiters.
+// It trims whitespace around each hostname and filters out empty strings.
+// Examples:
+//   - "host1.com,host2.com" -> ["host1.com", "host2.com"]
+//   - "host1.com, host2.com" -> ["host1.com", "host2.com"]
+//   - "host1.com\nhost2.com" -> ["host1.com", "host2.com"]
+//   - "host1.com,\nhost2.com" -> ["host1.com", "host2.com"]
+func SplitExternalHosts(s string) []string {
+	// First, replace newlines with commas to normalize the delimiter
+	normalized := strings.ReplaceAll(s, "\n", ",")
+	
+	// Split by comma
+	parts := strings.Split(normalized, ",")
```

Comment:
Please use https://pkg.go.dev/regexp@go1.25.3#Regexp.Split where the regex can then be ",|\n"